src/State/AbstractProcessor.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace App\State;
+
+use ApiPlatform\Metadata\DeleteOperationInterface;
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProcessorInterface;
+use Doctrine\ORM\EntityManagerInterface;
+
+abstract class AbstractProcessor implements ProcessorInterface
+{
+    public function __construct(
+        private readonly EntityManagerInterface $entityManager,
+        private readonly ProcessorInterface $persistProcessor,
+        private readonly ProcessorInterface $removeProcessor,
+        private readonly ?AbstractProvider $provider = null
+    ) {}
+
+    /**
+     * {@inheritdoc}
+     *
+     * @return T
+     */
+    public function process($data, Operation $operation, array $uriVariables = [], array $context = [])
+    {
+        if ($operation instanceof DeleteOperationInterface) {
+            return $this->removeProcessor->process($data, $operation, $uriVariables, $context);
+        }
+
+        $data = $this->fromInput($data, $operation, $uriVariables, $context);
+        $result = $this->persistProcessor->process($data, $operation, $uriVariables, $context);
+
+        return $this->toOutput($result);
+    }
+
+    /**
+     * @return T
+     */
+    protected function fromInput(mixed $object, Operation $operation, array $uriVariables, array $context): object
+    {
+        return $object;
+    }
+
+    public function toOutput(object $object): object
+    {
+        return null !== $this->provider ? $this->provider->toOutput($object) : $object;
+    }
+
+    /**
+     * Load previous object if any.
+     *
+     * This is needed to get an objet handled by entity manager.
```

Comment:
"object"

Reply:
Fixed in https://github.com/os2display/display-api-service/pull/173/commits/504d67ed15c0ce1a341af7d8147666e5ff6fcc74.